### PR TITLE
Improve error handling in connector: wrap with try catch

### PR DESCRIPTION
**Pull request by yuriy-fix**
Connected to https://github.com/vaadin/components-team-tasks/issues/479

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker-flow/51)

### CODE REVIEW

**Review by samie — COMMENTED** [review]
Is there technically a way to have a shared tryCatchWrapper for all the components?

**Comment by tomivirkki** [comment]
> Is there technically a way to have a shared tryCatchWrapper for all the components?

Should this be added to Flow directly, instead of repeating the same code in every connector? @pleku?

File: vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js [hunk 1]
```diff
@@ -1,236 +1,242 @@
...
-                if (timepicker.__inputElement.value !== newValue) {
-                    timepicker.__inputElement.value = newValue;
-                    timepicker.__dropdownElement.value = newValue;
-                    timepicker.__onInputChange();
-                }
-            }
-        }
-    }
-};
+            })
```
Comment (Haprog) [inline vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js]: This line should end in `;` as it is an assignment. This is the end of:
```js
timepicker.$connector.setLocale = tryCatchWrapper(function (locale) {
```

File: vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js [hunk 2]
```diff
@@ -1,236 +1,242 @@
...
+                const amString = getAmString(locale);
+
+                // 2. What is the separator ?
+                let localeTimeString = testPmTime.toLocaleTimeString(locale);
+                // since the next regex picks first non-number-whitespace, need to discard possible PM from beginning (eg. chinese locale)
+                if (pmString && localeTimeString.startsWith(pmString)) {
+                    localeTimeString = localeTimeString.replace(pmString, '');
+                }
+                const separator = localeTimeString.match(/[^\u0660-\u0669\s\d]/);
+                //console.info(locale.toUpperCase() + " AM/PM: " + amString + "/" + pmString + ", separator " + separator);
```
Comment (Haprog) [inline vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js]: I think this comment line should be removed (though it was not edited in this PR).

**Review by Haprog — CHANGES_REQUESTED** [review]
Mostly looks good but I think we could reduce the number of `tryCatchWrapper`s here.

To me it looks like everything is executed via one of these directly:
- `initLazy`
- `timepicker.$connector.setLocale`
- `formatTime`
- `parseTime`

I think everything would be covered even if we remove `tryCatchWrapper` from all other functions here except those 4 (and currently `formatTime` and `parseTime` aren't even wrapped with it but I think they should be).

So after adding the wrapper for `formatTime` and `parseTime` I think it's safe to remove it from these:
- `getAmPmString`
- `anyNumberCharToArabicNumberReplacer`
- `parseMillisecondCharsToInt`
- `getTimeFormatOptions`
- `formatMilliseconds`

Or what do you think?

File: vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js (hunk 1 above)
Comment (yuriy-fix) [reply vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js]: Updated.

File: vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js (hunk 2 above)
Comment (yuriy-fix) [reply vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js]: Removed.

**Comment by yuriy-fix** [comment]
@Haprog, thanks for good suggestions!
I was trying to avoid direct wrapping of `i18n` functions with `tryCatchWrapper` and instead wrap the relative methods in order not to mess-up those with wrappers from other components on the page when debugging.

But actually it makes sense, updated in latest commit.

**Review by Haprog — APPROVED** [review]

**Pull request merged**